File: 2021/Day 03 - Binary Diagnostic/Part1.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int main(){
    string l; 
    vector<string>lines;
    int n[15]  = {}, count = 0;
    while(cin >> l) {
        lines.push_back(l);
        count++;
    }

    long long g = 0, e = 0;
    for(int i = 0; i < l.size(); i++) {
        for(int j = 0; j < lines.size(); j++)
            n[i] += (lines[j][i] == '1');
        //cout << count << " " << n[i] << endl;
        if (n[i] > (count / 2)){
            g += (1 << (l.size() - 1 - i));
        } else {
            e += (1 << (l.size() - 1 - i));
        }
    }

    cout << 1LL * g * e;
    return 0;
}

File: 2023/Day 03- Gear Ratios/Part1.swift
import Foundation

func solve() {
    var lines: [String] = []
    while let line = readLine() {
        var aux = line
        aux.append(".")
        lines.append(aux)
    }
    var currentNumber = 0
    var isCurrentPar = false
    var ans = 0
    for i in 0 ..< lines.count {
        for j in 0 ..< lines[i].count {
            let c = Array(lines[i])[j]
            if (!c.isNumber) {
                if (isCurrentPar) {
                    ans += currentNumber
                }
                isCurrentPar = false
                currentNumber = 0
                continue
            }
            if let myNum = c.wholeNumberValue {
                currentNumber *= 10
                currentNumber += myNum
            }
            
            for k in -1 ... 1 {
                for l in -1 ... 1 {
                    let ni = i + k
                    let nj = j + l
                    if nj < 0 || ni < 0 || ni >= lines.count || nj >= lines[ni].count {
                        continue
                    }
                    let nc = Array(lines[ni])[nj]
                    if nc != "." {
                        if !nc.isNumber {
                            isCurrentPar = true
                        }
                    }
                }
            }
        }
    }
    print(ans)
}

solve()
